### PR TITLE
feat: add Kazakh locale

### PR DESCRIPTION
**Pull request by nodkz**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New locale addition

### 🔗 Related issue link

This PR awaiting for the following packages with just added KK_KZ locale:
- rc-pagination <https://github.com/react-component/pagination/pull/309>
- rc-picker <https://github.com/react-component/picker/pull/167>

Please approve the above PRs and bump its new package versions. Or you may merge this PR without awaiting packages above because the current PR is using the Russian language as a fallback (Kazakh people understand the Russian language).

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  🇰🇿 Add Kazakh locale.  |
| 🇨🇳 Chinese | 🇰🇿 新增哈萨克语。  |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered docs/react/i18n.en-US.md](https://github.com/nodkz/ant-design/blob/kk-locale/docs/react/i18n.en-US.md)

### CODE REVIEW

File: components/locale/kk_KZ.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  DatePicker,
+  TimePicker,
+  Calendar,
+  global: {
+    placeholder: 'Таңдаңыз',
+  },
+  Table: {
+    filterTitle: 'Сүзгі',
+    filterConfirm: 'Жарайды',
+    filterReset: 'Құлату',
```
Comment (manar-mk) [inline components/locale/kk_KZ.tsx:27]: I don't know the correct translation, but it's definitely not this one

**Comment by yoyo837** [comment]
Waiting for upstream release.

File: components/locale/kk_KZ.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+    filterEmptyText: 'Фильтр жоқ',
+    emptyText: 'Деректер жоқ',
+    selectAll: 'Барлығын таңдау',
+    selectInvert: 'Таңдауды төңкеру',
+    selectionAll: 'Барлық деректерді таңдаңыз',
+    sortTitle: 'Сұрыптау',
+    expand: 'Жолды кеңейту',
+    collapse: 'Жолды кеңейту',
```
Comment (roose) [inline components/locale/kk_KZ.tsx:35]: `expand:  'Жолды жазу'`, 
`collapse: 'Жолды бүктеу'`

File: components/locale/kk_KZ.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    okText: 'Жарайды',
+    cancelText: 'Болдырмау',
+  },
+  Transfer: {
+    titles: ['', ''],
+    searchPlaceholder: 'Іздеу',
+    itemUnit: 'элемент.',
+    itemsUnit: 'элемент.',
+    remove: 'Жою',
+    selectAll: 'Барлық деректерді таңдаңыз',
```
Comment (roose) [inline components/locale/kk_KZ.tsx:55]: `selectAll: 'Барлық деректерді таңдау'`

File: components/locale/kk_KZ.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    description: 'Деректер жоқ',
+  },
+  Icon: {
+    icon: 'белгішесі',
+  },
+  Text: {
+    edit: 'Өңдеу',
+    copy: 'Көшіру',
+    copied: 'Көшірілді',
+    expand: 'Кеңейту',
```
Comment (roose) [inline components/locale/kk_KZ.tsx:78]: `expand: 'Жазу'`

**Comment by yoyo837** [comment]
Need add `kk_KZ` local file test in `/components/locale-provider/__tests__/index.test.js`.

**Comment by afc163** [comment]
feature branch please

**Comment by nodkz** [comment]
changed base branch to `ant-design:feature`

**Comment by afc163** [comment]
Could you update https://github.com/ant-design/ant-design/edit/master/docs/react/i18n.zh-CN.md too, the chinese translation of Kazakh is `哈萨克语`.

**Review by afc163 — APPROVED** [review]

**Comment by yoyo837** [comment]
pagination和picker没发版

**Comment by nodkz** [comment]
I'll send a fix for picker & pagination (switch from Russian to Kazakh) when your package.json will be updated to new versions of rc-picker & rc-pagination with Kazakh locale.

Nothing to do right now. Thanks for merging 👍

**Comment by yoyo837** [comment]
You can update them with your own PR in the feature branch.

**Comment by nodkz** [comment]
@yoyo837 dependencies were updated in https://github.com/ant-design/ant-design/pull/27631

**Pull request merged**